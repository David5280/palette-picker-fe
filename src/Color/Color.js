import React from 'react'

const Color = (props) => {
  const { hexcode, index, handleLock, icon } = props;
  console.log(icon)
  return (
    <div 
      className='single-color'
      style= {
        {backgroundColor: `#${hexcode}`}
      }
    >
      <p>Color {index + 1}: </p>
      <p>#{hexcode.toUpperCase()}</p>
      <img 
        src={`${icon}`} 
        alt='locked-icon' 
        onClick={() => handleLock(hexcode)}
        className='locked-icon' 
      />
    </div>
  ) 
}

export default Color
