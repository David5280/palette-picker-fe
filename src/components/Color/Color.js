import React from 'react'

export const Color = (props) => {
  const { hexcode, index, handleLock, icon } = props;
  return (
    <div 
      className='single-color'
      style= {
        {backgroundColor: `#${hexcode}`}
      }
    >
    <div style={{borderRadius: '100px', margin:'auto', backgroundColor:'#'+hexcode, width:'100px'}}>
      <p className='color-info'>Color {index + 1}: </p>
      <p>#{hexcode.toUpperCase()}</p>
    </div>
      <img 
        src={`${icon}`} 
        alt='locked-icon' 
        onClick={() => handleLock(hexcode)}
        className='locked-icon color-info' 
      />
    </div>
  ) 
}

export default Color
