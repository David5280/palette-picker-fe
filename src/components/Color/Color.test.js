import React from 'react';
import ReactDOM from 'react-dom';
import Color from './Color';
import { shallow } from 'enzyme';

describe('Color Container', () => {
  let mockColorObj;
  let wrapper;
  beforeEach(() => {
    mockColorObj = {
      color: '#FFFEEE',
      paletteID: 4,
      dbKey: 'AYEE',
    }
    wrapper = shallow(<Color 
      color={mockColorObj.color}
      paletteID={mockColorObj.paletteID}
      dbKey={mockColorObj.dbKey}
      patchProject={jest.fn()}
      hexcode={'#ddbbcc'}
      index={1}
      handleLock={jest.fn()}
      icon={'/newIcon.png'}/>)
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  it('should call handleLock when the button is clicked', () => {
    wrapper.find('.locked-icon').simulate('click');
    expect(wrapper.instance().handleLock()).toHaveBeenCalled();
  });
});