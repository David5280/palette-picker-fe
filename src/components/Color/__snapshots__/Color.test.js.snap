// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Container should match the snapshot 1`] = `ShallowWrapper {}`;
